packages/react/src/components/DataTable/TableSelectRow.tsx

```diff
@@ -87,29 +87,58 @@ const TableSelectRow = ({
 }: TableSelectRowProps) => {
   const prefix = usePrefix();
   const uniqueNameId = useId();
+
+  const handleRadioChange = onChange
+    ? (
+        value: string | number | undefined,
+        name: string | undefined,
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        // Convert the radio value to boolean for consistency
+        onChange(!!value, name || '', event);
+      }
+    : undefined;
+
+  const handleCheckboxChange = onChange
+    ? (
+        checked: boolean,
+        name: string,
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        onChange(checked, name, event);
+      }
+    : undefined;
+
   const selectionInputProps = {
     id,
     name: name ? name : uniqueNameId,
     onClick: onSelect,
-    onChange,
     checked,
     disabled,
   };
-  const InlineInputComponent = radio ? RadioButton : InlineCheckbox;
+  // const InlineInputComponent = radio ? RadioButton : InlineCheckbox;
```

Comment:
Can we remove this commented out line? Or do we need it for reference still?
```suggestion
```

Reply:
Can removed it